mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td

```diff
@@ -676,6 +676,21 @@ def LLVM_DILabelAttr : LLVM_Attr<"DILabel", "di_label",
   let assemblyFormat = "`<` struct(params) `>`";
 }
 
+def LLVM_DIStringTypeAttr : LLVM_Attr<"DIStringType", "di_string_type",
+                                     /*traits=*/[], "DITypeAttr"> {
+  let parameters = (ins
+    LLVM_DITagParameter:$tag,
+    "StringAttr":$name,
+    OptionalParameter<"uint64_t">:$sizeInBits,
+    OptionalParameter<"uint64_t">:$alignInBits,
+    OptionalParameter<"DINodeAttr">:$stringLength,
```

Comment:
I think it would make sense to introduce DIVariableAttr in LLVMAttrs.h. Especially since this seems to exist in the LLVM class hierarchy.

When adding it is best to take DIScopeAttr, DITypeAttr, or a similar attribute as an example. There are a couple of places that need to be updated such as the DebugImport / DebugTranslation and LLVMAttrs.h / LLVMAttrs.cpp

Reply:
Thanks for the pointers. I have added `DIVariableAttr` and I think it makes the patch cleaner. With `DIVariableAttr` now available, there may be some cleanup possible in the `DISubrangeAttr` changes that I did last week. I will handle that in a separate PR.